engineering/python-style-guide.md

```diff
@@ -0,0 +1,13 @@
+source{d} Python Style Guide
+
+1. Contributing to existing projects - do not change the style.
+2. [PEP8](https://www.python.org/dev/peps/pep-0008/) with 99 char line length limit
+3. Indentation: 4 whitespaces
+4. Naming: classes: **CamelCase** functions: **lower_underscore_case** arguments: **lower_underscore_case**
+5. Class methods order: first public, then protected (_), then private (__)
+6. Use single or double quotes, but be consistent across the whole project. ML team use double quotes.
```

Comment:
PEP8 recommends to break consistency when needed to improve readability (something that has been commented but it's not written here): https://www.python.org/dev/peps/pep-0008/#string-quotes
**When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string.**